dataprofiler/data_readers/json_data.py

```diff
@@ -53,17 +55,37 @@ def __init__(self, input_file_path=None, data=None, options=None):
         #  _selected_keys: keys being selected from the entire dataset
         self._data_formats["records"] = self._get_data_as_records
         self._data_formats["json"] = self._get_data_as_json
+        self._data_formats["data_stream"] = self._get_data_as_data_stream
         self._selected_data_format = options.get("data_format", "dataframe")
-        self._intuitive_detection = options.get("intuitive_detection", False)
+        self._data_key = options.get("data_key", "data")
         self._selected_keys = options.get("selected_keys", list())
-
+        self._metadata = None
         if data is not None:
             self._load_data(data)
 
     @property
     def selected_keys(self):
         return self._selected_keys
 
+    @property
+    def metadata(self):
+        """
+        Returns a data frame that contains the metadata
+        """
+        if self._metadata is None:
+            warnings.warn("No metadata was detected")
+        return self._metadata
+
+    @property
+    def data_and_metadata(self):
+        """
+        Returns a data frame that joins the data and the metadata.
+        """
+        data = self.data
+        if self._metadata is not None:
+            data = [self._metadata, data]
+            data = pd.concat(data, axis=1)
+        return data
```

Comment:
This is an opinion, but i think returning as:
`{'data': self.data, 'metadata': self.metadata}`
might make more sense as it may more accurately represent original state

Reply:
Also, metadata key or data key (payload) etc we should think about adding as an option in a future PR

Reply:
i see data is an option, metadata could be as well?